core/src/main/java/org/nervos/ckb/transaction/TxGenerator.java

```diff
@@ -19,40 +21,76 @@
+  private SystemScriptCell systemSecpCell;
   private CKBService ckbService;
 
-  public TxGenerator(String privateKey, CKBService ckbService) {
-    this.privateKey = privateKey;
+  public TxGenerator(CKBService ckbService) {
     this.ckbService = ckbService;
 
     try {
-      this.systemScriptCell = Utils.getSystemScriptCell(ckbService);
+      this.systemSecpCell = Utils.getSystemScriptCell(ckbService);
     } catch (Exception e) {
       e.printStackTrace();
     }
-    this.lockScript = Utils.generateLockScriptWithPrivateKey(privateKey, systemScriptCell.cellHash);
   }
 
-  public Transaction generateTx(List<Receiver> receivers) throws IOException {
+  public Transaction generateTx(String privateKey, List<Receiver> receivers) throws IOException {
     BigInteger needCapacities = BigInteger.ZERO;
     for (Receiver receiver : receivers) {
       needCapacities = needCapacities.add(receiver.capacity);
     }
-    if (needCapacities.compareTo(new BigInteger(MIN_CAPACITY)) < 0) {
+    Sender sender = new Sender(privateKey, needCapacities);
+    return generateTx(Collections.singletonList(sender), receivers);
+  }
+
+  public Transaction generateTx(List<Sender> senders, List<Receiver> receivers) throws IOException {
+    BigInteger needCapacities = BigInteger.ZERO;
+    for (Receiver receiver : receivers) {
+      needCapacities = needCapacities.add(receiver.capacity);
+    }
+    if (needCapacities.compareTo(MIN_CAPACITY) < 0) {
       throw new IOException("Less than min capacity");
     }
 
-    Cells cellInputs =
-        new CellGatherer(ckbService).getCellInputs(lockScript.computeHash(), needCapacities);
-    if (cellInputs.capacity.compareTo(needCapacities) < 0) {
-      throw new IOException("No enough Capacities");
+    BigInteger sendCapacities = BigInteger.ZERO;
+    for (Sender sender : senders) {
+      sendCapacities = sendCapacities.add(sender.capacity);
+    }
+    if (sendCapacities.compareTo(needCapacities) < 0) {
+      throw new IOException("Sender capacity amount less than receiver capacity amount");
     }
 
+    // gather cell inputs through senders' addresses
+    // and put cellInput and related private key to a list
+    BigInteger collectCapacity = BigInteger.ZERO;
+    List<Transaction.CellWithPrivateKey> cellWithPrivateKeys = new ArrayList<>();
+    List<CellInput> cellInputs = new ArrayList<>();
+    List<Witness> witnesses = new ArrayList<>();
+    for (Sender sender : senders) {
+      String lockHash =
+          Utils.generateLockScriptWithPrivateKey(sender.privateKey, systemSecpCell.cellHash)
+              .computeHash();
+      CollectedCells collectedCells =
+          new CellGatherer(ckbService).getCellInputs(lockHash, sender.capacity);
+      if (collectedCells.capacity.compareTo(sender.capacity) < 0) {
+        throw new IOException("No enough Capacities with sender private key: " + sender.privateKey);
+      }
+      collectCapacity = collectCapacity.add(collectedCells.capacity);
```

Comment:
collectedCapacity may be better?

Reply:
Updated. https://github.com/nervosnetwork/ckb-sdk-java/pull/170/commits/7ad9f29f89b60ae4cb3cd7d527edafcf66c88c8e